File: frontend/src/modules/repository/application/use-cases/readRepoSignature.ts
import { RepoSignatureReader } from "@modules/repository/application/contracts/RepoSignatureReader";

export async function readRepoSignature(
  repoSignatureReader: RepoSignatureReader,
  repositoryPath: string
): Promise<string> {
  return repoSignatureReader.readRepoSignature(repositoryPath);
}

File: backend/src/app/repo_signature/read_repo_signature.rs
use std::path::Path;

use crate::app::repo_signature::contracts::RepoSignatureReader;
use crate::domain::errors::AppError;

pub fn read_repo_signature(
    repo_signature_reader: &dyn RepoSignatureReader,
    repository_path: &Path,
) -> Result<String, AppError> {
    repo_signature_reader.read_signature(repository_path)
}
